### PR TITLE
fix(services-endorsements-api): Keep metadata up to date

### PR DESCRIPTION
**Pull request by stefanhar**
# Keep metadata up to date

There is a new requirement for the system to keep voter region and addresses up to date for endorsers.

## What

- Added new metadata field to endorsementLists with keepUpToDate option
- Added update metadata script
- Updated applications depending on the endorsement system
- Updated seeds to provide more realistic test data
- Updated tests to reflect new functionality
- Made passing auth token optional to allow for system calls to getMetadata of metadata service

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by alexandradilja — APPROVED** [review]

File: apps/services/endorsements/api/infra/endorsement-system-scripts-update-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { ref, service, ServiceBuilder } from '../../../../../infra/src/dsl/dsl'
+import { settings } from '../../../../../infra/src/dsl/settings'
+
+export const serviceSetup = (services: {
+  servicesTemporaryVoterRegistryApi: ServiceBuilder<'services-temporary-voter-registry-api'>
+}): ServiceBuilder<'endorsement-system-scripts-update-metadata'> =>
+  service('endorsement-system-scripts-update-metadata')
+    .namespace('endorsement-system')
+    .image('services-endorsements-scripts-update-metadata')
+    .serviceAccount()
```
Comment (pshomov) [inline apps/services/endorsements/api/infra/endorsement-system-scripts-update-metadata.ts:10]: you should remove this line, we do not need service account for this one

Comment (stefanhar) [reply apps/services/endorsements/api/infra/endorsement-system-scripts-update-metadata.ts]: Removed

File: apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts [hunk 2]
```diff
@@ -33,9 +33,10 @@ export class VoterRegistryController {
   }
 
   @ApiOkResponse({
-    description: 'Finds voter registry entry given user authentication',
+    description: 'Finds voter registry entry given users national id',
     type: VoterRegistry,
   })
+  @BypassAuth() // currently we have no way to create system auth tokens, we will add those when available for better tracking
```
Comment (eirikurn) [inline apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts:39]: Hm? This should be possible, by using client credentials to get an access token with some endorsement client and system scope. 

Please follow this up with a PR to fix this.

**Review by eirikurn — APPROVED** [review]

File: apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts (hunk 2 above)
Comment (stefanhar) [reply apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts:39]: Hmm, I might not familiar enough with the auth library 😅 
This is initialized via cron job that uses the metadata service to get the data.
Are we able to create a system token without user credentials? Though this was not yet implemented?

Comment (eirikurn) [reply apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts:39]: Yes, this is built in functionality in Identity Server. I'm not sure if we have any reference code or helpers yet, but Fuglar can provide implementation details.

Comment (stefanhar) [reply apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts:39]: Nice, will contact them about this ❤️

Comment (eirikurn) [reply apps/services/temporary-voter-registry-api/src/app/modules/voterRegistry/voterRegistry.controller.ts:39]: Thank you ❤️

**Comment by stefanhar** [comment]
Merged in https://github.com/island-is/island.is/pull/4871

**Pull request closed without merging**